criu/cgroup.c

```diff
@@ -716,32 +716,31 @@ static int collect_cgroups(struct list_head *ctls)
 	return 0;
 }
 
-int dump_task_cgroup(struct pstree_item *item, u32 *cg_id, struct parasite_dump_cgroup_args *args)
+int dump_task_cgroup(struct pid *item_pid, u32 *cg_id, struct parasite_dump_cgroup_args *args)
 {
 	int pid;
 	LIST_HEAD(ctls);
 	unsigned int n_ctls = 0;
 	struct cg_set *cs;
 
-	if (item)
-		pid = item->pid->real;
+	if (item_pid)
+		pid = item_pid->real;
 	else
 		pid = getpid();
-
 	pr_info("Dumping cgroups for %d\n", pid);
 	if (parse_task_cgroup(pid, args, &ctls, &n_ctls))
 		return -1;
 
-	cs = get_cg_set(&ctls, n_ctls, item);
+	cs = get_cg_set(&ctls, n_ctls, item_pid);
```

Comment:
`item_pid` was previously local variable `pid`, which caused the cgroups02 test to fail -- `pid` would prevent the `get_cg_set` function from fully dumping the cgroups.